File: public/docs/_examples/toh-4/ts/app/app.component.1.ts
// #docplaster
// #docregion on-init
import { OnInit } from '@angular/core';

// #enddocregion on-init
import { Component } from '@angular/core';

import { Hero } from './hero';
import { HeroDetailComponent } from './hero-detail.component';
// #docregion hero-service-import
import { HeroService } from './hero.service.1';
// #enddocregion hero-service-import

// Testable but never shown
@Component({
  selector: 'my-app',
  template: `
  <div *ngFor="let hero of heroes" (click)="onSelect(hero)">
    {{hero.name}}
  </div>
  <my-hero-detail [hero]="selectedHero"></my-hero-detail>
  `,
  directives: [HeroDetailComponent],
  // #docregion providers
  providers: [HeroService]
  // #enddocregion providers
})
// #docregion on-init
export class AppComponent implements OnInit {
  // #enddocregion on-init
  title = 'Tour of Heroes';
  // #docregion heroes-prop
  heroes: Hero[];
  // #enddocregion heroes-prop
  selectedHero: Hero;

  /*
  // #docregion new-service
  heroService = new HeroService(); // don't do this
  // #enddocregion new-service
  */
  // #docregion ctor
  constructor(private heroService: HeroService) { }
  // #enddocregion ctor
  // #docregion getHeroes
  getHeroes() {
    // #docregion get-heroes
    this.heroes = this.heroService.getHeroes();
    // #enddocregion get-heroes
  }
  // #enddocregion getHeroes

  // #docregion ng-on-init
  // #docregion on-init
  ngOnInit() {
    // #enddocregion on-init
    this.getHeroes();
    // #docregion on-init
  }
  // #enddocregion on-init
  // #enddocregion ng-on-init

  onSelect(hero: Hero) { this.selectedHero = hero; }
  // #docregion on-init
}

File: public/docs/_examples/toh-4/dart/lib/app_component_1.dart
// #docplaster
// #docregion on-init
import 'package:angular2/core.dart';

// #enddocregion on-init
import 'hero.dart';
import 'hero_detail_component.dart';
import 'hero_service_1.dart';

// Testable but never shown
@Component(
    selector: 'my-app',
    template: '''
    <div *ngFor="let hero of heroes" (click)="onSelect(hero)">
      {{hero.name}}
    </div>
    <my-hero-detail [hero]="selectedHero"></my-hero-detail>
    ''',
    directives: const [HeroDetailComponent],
    // #docregion providers
    providers: const [HeroService])
    // #enddocregion providers
// #docregion on-init
class AppComponent implements OnInit {
  // #enddocregion on-init
  String title = 'Tour of Heroes';
  // #docregion heroes-prop
  List<Hero> heroes;
  // #enddocregion heroes-prop
  Hero selectedHero;

  // #docregion new-service
  HeroService heroService = new HeroService(); // don't do this
  // #enddocregion new-service
  // #docregion ctor
  final HeroService _heroService;
  AppComponent(this._heroService);
  // #enddocregion ctor

  // #docregion getHeroes
  void getHeroes() {
    // #docregion get-heroes
    heroes = _heroService.getHeroes();
    // #enddocregion get-heroes
  }
  // #enddocregion getHeroes

  // #docregion ng-on-init, on-init
  void ngOnInit() {
    // #enddocregion on-init
    getHeroes();
    // #docregion on-init
  }
  // #enddocregion ng-on-init, on-init

  void onSelect(Hero hero) {
    selectedHero = hero;
  }
  // #docregion on-init
}
